plugins/state_history_plugin/include/eosio/state_history_plugin/session.hpp

```diff
@@ -16,691 +19,327 @@ extern const char* const state_history_plugin_abi;
-      fc_dlog(plugin.get_logger(), "received get_blocks_ack_request_v0 = ${req}", ("req", req));
-      if (!current_request) {
-         fc_dlog(plugin.get_logger(), " no current get_blocks_request_v0, discarding the get_blocks_ack_request_v0");
-         return;
+      try {
+         co_await f();
       }
-
-      auto self = this->shared_from_this();
-      auto entry_ptr = std::make_unique<blocks_ack_request_send_queue_entry<session>>(self, std::move(req));
-      session_mgr.add_send_queue(std::move(self), std::move(entry_ptr));
-   }
-
-   state_history::get_status_result_v0 get_status_result() {
-      fc_dlog(plugin.get_logger(), "replying get_status_request_v0");
-      state_history::get_status_result_v0 result;
-      result.head              = plugin.get_block_head();
-      result.last_irreversible = plugin.get_last_irreversible();
-      result.chain_id          = plugin.get_chain_id();
-      auto&& trace_log         = plugin.get_trace_log();
-      if (trace_log) {
-         auto r = trace_log->block_range();
-         result.trace_begin_block = r.first;
-         result.trace_end_block   = r.second;
+      catch(std::bad_alloc&) {
+         throw;
       }
-      auto&& chain_state_log = plugin.get_chain_state_log();
-      if (chain_state_log) {
-         auto r = chain_state_log->block_range();
-         result.chain_state_begin_block = r.first;
-         result.chain_state_end_block   = r.second;
+      catch(fc::exception& e) {
+         if(has_logged_exception.test_and_set() == false)
+            fc_ilog(logger, "state history connection from ${a} failed: ${w}", ("a", remote_endpoint_string)("w", e.top_message()));
       }
-      fc_dlog(plugin.get_logger(), "pushing get_status_result_v0 to send queue");
-
-      return result;
-   }
-
-   void update_current_request_impl(state_history::get_blocks_request_v0& req) {
-      fc_dlog(plugin.get_logger(), "replying get_blocks_request_v0 = ${req}", ("req", req));
-      to_send_block_num = std::max(req.start_block_num, plugin.get_first_available_block_num());
-      for (auto& cp : req.have_positions) {
-         if (req.start_block_num <= cp.block_num)
-            continue;
-         auto id = plugin.get_block_id(cp.block_num);
-         if (!id || *id != cp.block_id)
-            req.start_block_num = std::min(req.start_block_num, cp.block_num);
-
-         if (!id) {
-            to_send_block_num = std::min(to_send_block_num, cp.block_num);
-            fc_dlog(plugin.get_logger(), "block ${block_num} is not available", ("block_num", cp.block_num));
-         } else if (*id != cp.block_id) {
-            to_send_block_num = std::min(to_send_block_num, cp.block_num);
-            fc_dlog(plugin.get_logger(), "the id for block ${block_num} in block request have_positions does not match the existing",
-                    ("block_num", cp.block_num));
-         }
+      catch(boost::system::system_error& e) {
```

Comment:
fwiw, I broke out `boost::system::system_error` to be treated specially because the `what()` on it contains a lot of information -- even the filename. It's the difference between
```
state history connection from 127.0.0.1:55066 failed: End of file
```
vs
```
state history connection from 127.0.0.1:34910 failed: End of file [asio.misc:2 at /home/foo/leap/libraries/boost/libs/asio/include/boost/asio/detail/reactive_socket_recv_op.hpp:134:5 in function 'static void boost::asio::detail::reactive_socket_recv_op<MutableBufferSequence, Handler, IoExecutor>::do_complete(void*, boost::asio::detail::operation*, const boost::system::error_code&, std::size_t)']
```